device/WakeOnLan/controller.js

```diff
@@ -0,0 +1,157 @@
+﻿'use strict';
+
+const BluePromise = require('bluebird');
+const fs = require('fs')
+const readline = require('readline')
+const os = require('os');
+const net = require('net');
+const dgram = require('dgram');
+
+const PORT_NUMBER = 9;
+const DEFAULT_IPADDRESS = '255.255.255.255';
+
+let wolMap = "wol.map";
+
+/**
+ * Sets the path to the 'wol.map' file
+ */
+module.exports.setMapPath = function (mapPath) {
+    wolMap = mapPath;
+}
+
+/**
+ * Handles button presses for WOL
+ */
+module.exports.onButtonPressed = function (deviceid, name) {
+    console.log(`WOL button pressed for ${name}`);
+
+    parseAddr(name)
+        .then(function (addr) {
```

Comment:
Nice - about time they implemented something like that (I haven't even looked at the ES6 spec in detail - so hints like this are great!)